client/carwash.lua

```diff
@@ -6,61 +6,57 @@ local function DrawText3Ds(x, y, z, text)
     SetTextFont(4)
     SetTextProportional(true)
     SetTextColour(255, 255, 255, 215)
-    SetTextEntry("STRING")
+    BeginTextCommandDisplayText('STRING')
     SetTextCentre(true)
-    AddTextComponentString(text)
+    AddTextComponentSubstringPlayerName(text)
     SetDrawOrigin(x,y,z, 0)
-    DrawText(0.0, 0.0)
+    EndTextCommandDisplayText(0.0, 0.0)
     local factor = (string.len(text)) / 370
     DrawRect(0.0, 0.0+0.0125, 0.017+ factor, 0.03, 0, 0, 0, 75)
     ClearDrawOrigin()
 end
 
 RegisterNetEvent('qb-carwash:client:washCar', function()
-    local PlayerPed = PlayerPedId()
-    local PedVehicle = GetVehiclePedIsIn(PlayerPed, false)
     washingVehicle = true
-    QBCore.Functions.Progressbar("search_cabin", "Vehicle is being washed ..", math.random(4000, 8000), false, true, {
+    QBCore.Functions.Progressbar('search_cabin', 'Vehicle is being washed ..', math.random(4000, 8000), false, true, {
         disableMovement = true,
         disableCarMovement = true,
         disableMouse = false,
         disableCombat = true,
     }, {}, {}, {}, function() -- Done
-        SetVehicleDirtLevel(PedVehicle, 0.0)
-        SetVehicleUndriveable(PedVehicle, false)
-        WashDecalsFromVehicle(PedVehicle, 1.0)
+        SetVehicleDirtLevel(cache.vehicle, 0.0)
+        SetVehicleUndriveable(cache.vehicle, false)
+        WashDecalsFromVehicle(cache.vehicle, 1.0)
         washingVehicle = false
     end, function() -- Cancel
-        QBCore.Functions.Notify("Washing canceled ..", "error")
+        QBCore.Functions.Notify('Washing canceled ..', 'error')
         washingVehicle = false
     end)
 end)
 
 CreateThread(function()
     local sleep
     while true do
-        local PlayerPed = PlayerPedId()
-        local PlayerPos = GetEntityCoords(PlayerPed)
-        local PedVehicle = GetVehiclePedIsIn(PlayerPed, false)
-        local Driver = GetPedInVehicleSeat(PedVehicle, -1)
-        local dirtLevel = GetVehicleDirtLevel(PedVehicle)
+        local PlayerPos = GetEntityCoords(cache.ped)
+        local Driver = GetPedInVehicleSeat(cache.vehicle, -1)
```

Comment:
Convert this to `local Driver = cache.seat == -1 and cache.ped or 0` and then just check if Driver is not 0 in the loop

Reply:
Done